File: src/main/scala/actors5/Servidor.scala
package hello.actors5

import java.net.{ServerSocket,Socket}

case class Saluda(nombre:String, fecha:java.util.Date, socket:Socket)

object Servidor {

  def run(port:Int) = {
    val server = new ServerSocket(port)
    //Tenemos que iniciar estos a patin
    Lector.start
    Escritor.start
    while (true) {
      val sock = server.accept
      println("Nueva conexion")
      Lector ! sock
    }
  }

  def main(args:Array[String]) = {
    val port = if (args.length > 0) args(0).toInt else 9999
    println("Corriendo servidor en puerto " + port)
    run(port)
  }

}

File: src/main/java/hola/Servidor.java
package hola;
import java.net.*;
import java.io.*;

public class Servidor {

  public static void main(String[] args) {
    try {
      ServerSocket server = new ServerSocket(9999);
      while (true) {
        Socket socket = server.accept();
        //Por ahora dejemosle la chamba de los hilos al lector
        Lector.leer(socket);
      }
    } catch (IOException ex) {
      System.out.println("Ha tronado!");
    }
  }
}
